compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/CfgTraverser.kt

```diff
@@ -75,3 +80,60 @@ private fun <I : ControlFlowInfo<I, K, V>, K : Any, V : Any> ControlFlowGraph.co
         }
     }
 }
+
+// ---------------------- Path-sensitive data collection -----------------------
+
+fun <I : ControlFlowInfo<I, K, V>, K : Any?, V : Any> ControlFlowGraph.collectPathAwareDataForNode(
```

Comment:
Actually all control flow analysises should be path aware, so it's good to totally replace `collectDataForNode` with `collectPathAwareDataForNode` (and rename last one to `collectDataForNode`)

Reply:
Okay... good to hear that we want all CFA checkers to be path-sensitive. However, augmenting path info to existing control-flow info implies changes on how each checker interprets the collected info. It's unclear how easy or non-trivial it would be to migrate all analyses. Can I do that as a follow-up PR? I'd like to tackle one by one, or certain number/group of checkers.